chirp/drivers/uvk5.py

```diff
@@ -2083,3 +2064,19 @@ class RA79Radio(UVK5Radio):
     """Retevis RA79"""
     VENDOR = "Retevis"
     MODEL = "RA79"
+
+
+# This is from a radio i purchased from AliExperess store "Eagle Store"
+# which was quoted as both "Quansheng UV-K6" (in title)
+# and "Quansheng UV-K5(99)" in the imagery.
+# I'm not 100% sure it's legit Quansheng!
+
+
+@directory.register
+class UVK5_99_Radio(UVK5Radio):
+    MODEL = "UV-K5(99)"
+
+    @classmethod
+    def k5_approve_firmware(cls, firmware):
+        approved_prefixes = ('OSFW-bd90ca3')
```

Comment:
If there is no difference between the models, there is no need for a separate subclass, just add this to the list in the base class. Given that it's already in the alias list as such, that's what people are already expecting.

Also, it looks like the suffix (after the dash) is a git hash, which means it's likely to change. So probably just add `OSFW` to that list. Unfortunately, that probably also means it's going to be unstable in the future, but that seems to be the destiny of this model :/

Reply:
Be aware that the hardware is the same, but some vendors are likely now shipping radios with a specific build of fagci-mod firmware.

It's documented here by armel (F4HWN): https://www.youtube.com/watch?v=mhgHB2H6n34

TLDW: The OSFW bit comes from https://github.com/fagci/uv-k5-firmware-fagci-mod
The bd90ca3 bit comes from the hash for a specific commit here: https://github.com/fagci/uv-k5-firmware-fagci-mod/tree/bd90ca3f1621204ca2c17f21f0d1fb6ef5caea8d

So it may not be directly compatible with the original UV-K5.

Reply:
Yep, I'm aware, which is why I added this firmware checking and detection system for the UV-K5. I'm assuming the submitter has tested the OSFW stuff with the current driver since they submitted it with no delta from the base. If there is a difference, then we definitely need a subclass, but one that makes changes from the base class, like `uvk5_egzumer.py` does. And, it needs to be `@detected_by` the base class and not just `@register`'d so it doesn't show up as its own model.

Reply:
i will do some reading, some more testing, and resubmit. Fab resources ty.